apps/crn-server/test/handlers/research-output/index-handler.test.ts

```diff
@@ -19,30 +20,52 @@ describe('Research Output index handler', () => {
 
   test('Should fetch the research-output and create a record in Algolia when research-output is created', async () => {
     const researchOutputResponse = getResearchOutputResponse();
-    researchOutputControllerMock.fetchById.mockResolvedValueOnce(
-      researchOutputResponse,
-    );
+    const ownRelatedResearchOutput = getRelatedResearchOutputResponse(true);
+    const notOwnRelatedResearchOutput = getRelatedResearchOutputResponse(false);
```

Comment:
you should ready my comment about the fixture generator first, because I think that + this code here show that you have done something that was not indended.

If I understand correctly you have a research output here which we fetch from the data provider then save in algolia, but then should you not simply look at the `relatedResearch` and index each one of them regardless of the `isOwnRelatedResearchLink` (which you are doing, according to implementation)? So your tests should then modify the `isOwnRelatedResearchLink` property on the elements of the `relatedResearch` array of the `researchOutputResponse` response, whereas you seem to be returning the releated research from the data provider mock, but that never happens (and like I said this only worked cause typescript is not strict enough to catch this bug)

Reply:
Updated accordingly